backend/auth/middleware.go

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2024 Canonical Ltd.
+
+package auth
+
+import (
+	"crypto/rand"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/golang-jwt/jwt"
+	"github.com/omec-project/webconsole/backend/logger"
+)
+
+const (
+	UserRole  = 0
+	AdminRole = 1
+)
+
+type jwtGocertClaims struct {
```

Comment:
```suggestion
type jwtWebconsoleClaims struct {
```
Gocert is irrelavant here. Suggesting to use a name which is compatible with project.